8307804: Reorganize ArrayJuggle test cases

lkorinth opened the pull request:
Move all ArrayJuggle test cases to the same directory: test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle

Rename Juggle01 to Juggle3 (so it will not be confused with Juggle1)

Remove all directories and files used to launch the tests, instead use multiple  `@test id=xx` "annotations" in the four kept test files.

Create a new test file Juggle3Quick.java that will act as a quick group of tests. Unfortunately `#id` selectors can not be used in test groups so this is a workaround. See: https://bugs.openjdk.org/browse/CODETOOLS-7903467

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307804](https://bugs.openjdk.org/browse/JDK-8307804): Reorganize ArrayJuggle test cases


### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13929/head:pull/13929` \
`$ git checkout pull/13929`

Update a local copy of the PR: \
`$ git checkout pull/13929` \
`$ git pull https://git.openjdk.org/jdk.git pull/13929/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13929`

View PR using the GUI difftool: \
`$ git pr show -t 13929`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13929.diff">https://git.openjdk.org/jdk/pull/13929.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13929#issuecomment-1543855439)

lkorinth commented:
/label add hotspot-gc

dholmes-ora reviewed (commented):
Changes are fine in principle. I haven't tried to verify the details of each test case.

I've made a number of comments below about reformatting the `@test` segments to the normal multi-line format. In the PR I found these very hard to read ( Ididn't even realize jtreg would process them as a single line like that!). I did discover afterwards that these look much better when the file is viewed wide-screen so I will leave it for GC reviewers to decide what they prefer.

Thanks.

lkorinth commented on test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle/Juggle3.java:
```diff
@@ -0,0 +1,125 @@
+ * VM Testbase keywords: [gc, stress, stressopt, nonconcurrent]
+ */
+
+// Run in Juggle3Quic.java @test id=1  @key stress randomness @library /vmTestbase /test/lib @run main/othervm -XX:+HeapDumpOnOutOfMemoryError -Xlog:gc=debug:gc.log gc.ArrayJuggle.Juggle3 -gp byteArr -ms low
```
Yes, it is a comment, they show the quickgroup. Unfortunately I can not run those tests directly from a group and I needed to create Juggle3Quic.java (https://bugs.openjdk.org/browse/CODETOOLS-7903467). If you prefer I will remove those comments.

lkorinth commented on test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle/ArrayJuggle.README:
Yes, nice catch! It was not up to date to begin with. Although the description of some parts are still correct --- for me --- the README adds little benefit, and I would prefer removing the file. Another option is to keep lines up to and including line 51, and remove the rest. These kind of files just tend to bit rot.

lkorinth replied:
(of course also remove -- These tests run forever at the current time [8/14/97] --)

lkorinth commented:
Regarding multi line format, I think there is a strong value of being able to "scan" line after line seeing what permutations are tested. I recently fixed [JDK-8306435](https://bugs.openjdk.org/browse/JDK-8306435) --- such bugs would be much harder to make if we can easily "scan" the permutations, and was one reason why I choose to reorganise the tests.

lmesnik commented on test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle/Juggle3.java:
```diff
@@ -0,0 +1,125 @@
+ */
+
+// Run in Juggle3Quic.java @test id=1  @key stress randomness @library /vmTestbase /test/lib @run main/othervm -XX:+HeapDumpOnOutOfMemoryError -Xlog:gc=debug:gc.log gc.ArrayJuggle.Juggle3 -gp byteArr -ms low
+/* @test id=2  @key stress randomness @library /vmTestbase /test/lib @run main/othervm -XX:+HeapDumpOnOutOfMemoryError -Xlog:gc=debug:gc.log gc.ArrayJuggle.Juggle3 -gp byteArr -ms medium */
```
It would be much better to have a meaningful id like 'gc_byteArr_ms_medium'. So we can easier identify failures and easily add/remove rearrange testcases.

lkorinth replied:
I added IDs with names although a bit shorter.

lmesnik reviewed (commented):
Thanks for this clean up. There are few comments about names.

lkorinth commented:
I removed the test cases that were commented out. I added descriptive ids to the test cases (although they are not used now, they might be used in the future when they could be used to create a quick test group), and I removed the readme that I though was of little help and since long not updated.

dholmes-ora reviewed (approved):
Nothing further from me.

Thanks.

lkorinth commented:
Thanks David and Leonid! I will integrate after the weekend.
